src/result.js

```diff
@@ -180,7 +180,9 @@ class Result {
       connectionHolder.getConnection().then(
         // onFulfilled:
         connection => {
-          release(connection ? connection.protocol().version : undefined)
+          release(
```

Comment:
Why should connection be undefined here?
In onRejected at 188 I can see that there is no connection but that one still calls release. Why?

Reply:
The release is completing the result and release the connection in some cases. It's doing two things. I think it could be splited.

About checking the connection, the get connection should not resolve it a null connection, so this check should could be removed.